cmake/compiler/gcc/target.cmake

```diff
@@ -110,7 +110,6 @@ get_filename_component(LIBGCC_DIR ${LIBGCC_FILE_NAME} DIRECTORY)
 assert_exists(LIBGCC_DIR)
 
 LIST(APPEND LIB_INCLUDE_DIR "-L\"${LIBGCC_DIR}\"")
```

Comment:
It seems like these two values need to go together? Why is LIBGCC_DIR being handled here while libgcc itself is moved elsewhere?

Reply:
thanks for spotting.
It probably should, especially as the usage of `LIB_INCLUDE_DIR` was adjusted for clang so will take a look.

As for the `gcc` move, then that's moved because any value which is generally pre-known by the toolchain (`-lgcc`) and therefore should be present in a `linker_libraries.cmake` file, but the actual path is not known at this time.
The path is fetched by querying the compiler, and that cannot be done until the target arch (and thus the correct target compiler) has been decided.

That said, generally it should not be necessary to first query the compiler about `--print-libgcc-file-name` just to feed this information back to the linker later. Specifically for Zephyr SDK this is in principle not needed.

Where customizing `-L` / `LIB_INCLUDE_DIR` makes most sense is when SYSROOT_DIR is used to direct the linker to specific library location, as well as the `cross-compile` toolchain usage.
The core of the current implementation originates back to the KBuild transition days, https://github.com/zephyrproject-rtos/zephyr/commit/12f8f7616567f224b920512b6f81a65421e30bae.
However, revisiting current implementation is outside scope of this PR.

Reply:
That all sounds reasonable; feel free to leave this as-is for now; I'll be re-working #65434 once this has landed (and I have enough spare time).

Reply:
aligned to use linker properties.